### ISSUE TITLE
Rebase 33 to 34, no sound devices; failed to update rpmdb

### ISSUE THREAD

**Issue by bnordgren** [fedora-silverblue/issue-tracker#160]
After I rebased from 33 to 34, there are no sound devices available in the GNOME control panel.  I know there is a change to Pipewire on this release, but I have no idea how to troubleshoot. My aplay -l yields: 

```
[bnordgren@mine ~]$ aplay -l 
**** List of PLAYBACK Hardware Devices ****
card 0: USB [Scarlett 2i2 USB], device 0: USB Audio [USB Audio]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: HDMI [HDA Intel HDMI], device 3: HDMI 0 [HDMI 0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: HDMI [HDA Intel HDMI], device 7: HDMI 1 [HDMI 1]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: HDMI [HDA Intel HDMI], device 8: HDMI 2 [HDMI 2]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: HDMI [HDA Intel HDMI], device 9: HDMI 3 [HDMI 3]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: HDMI [HDA Intel HDMI], device 10: HDMI 4 [HDMI 4]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 2: PCH [HDA Intel PCH], device 0: ALC221 Analog [ALC221 Analog]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
```

Rebasing a different machine did not yield this problem.

**Comment by tpopela** [comment]
This isn't anything specific to Silverblue - can you please open an issue in https://gitlab.freedesktop.org/pipewire/pipewire ?

**Comment by bnordgren** [comment]
Actually, this is my work computer, which I use for virtual meetings. No sound is a showstopper so I'm rebasing back to F33. I can't be wrapped up in back-and-forth between upstream and the distribution packagers. Can you at least tell me how to opt out of pipewire using fedora 34, or offer something to rebase against which does not use it?

**Comment by bnordgren** [comment]
Well, there's other, more silverblue-specific problems too, I see, preventing the rebase downgrade from occurring...

This is a very strange error...

```
[bnordgren@mine ~]$ rpm-ostree rebase fedora:fedora/33/x86_64/silverblue
2 metadata, 0 content objects fetched; 788 B transferred in 2 seconds; 0 bytes content written
Checking out tree aa351a9... done
Inactive requests:
  opensc (already provided by opensc-0.21.0-1.fc33.x86_64)
Enabled rpm-md repositories: updates fedora google-chrome fedora-cisco-openh264 updates-archive
Updating metadata for 'updates'... done
rpm-md repo 'updates'; generated: 2021-05-06T00:44:52Z
Updating metadata for 'fedora'... done
rpm-md repo 'fedora'; generated: 2020-10-19T23:27:19Z
Updating metadata for 'google-chrome'... done
rpm-md repo 'google-chrome'; generated: 2021-05-05T17:49:20Z
Updating metadata for 'fedora-cisco-openh264'... done
rpm-md repo 'fedora-cisco-openh264'; generated: 2020-08-25T19:10:34Z
Updating metadata for 'updates-archive'... done
rpm-md repo 'updates-archive'; generated: 2021-05-06T01:52:46Z
Importing rpm-md... done
Resolving dependencies... done
Will download: 149 packages (79.3 MB)
Downloading from 'updates'... done
Downloading from 'fedora'... done
Importing packages... done
Checking out packages... done
Running pre scripts... done
Running post scripts... done
Running posttrans scripts... done
Writing rpmdb... done
error: Failed to update rpmdb (rpmtsRun code -1)
```

**Comment by bnordgren** [comment]
`rpm-ostree update` behaves normally. Rebasing does not. I tried rebasing back to 33 as well as 32. Same `rpmts code -1` error.

**Comment by bnordgren** [comment]
Since I can't downgrade to something that works, I did put in an issue on the pipewire issue tracker: 

https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/1155

**Comment by tpopela** [comment]
> `rpm-ostree update` behaves normally. Rebasing does not. I tried rebasing back to 33 as well as 32. Same `rpmts code -1` error.

This is https://github.com/fedora-silverblue/issue-tracker/issues/158 - we had to response from rpm-ostree devs so far.

**Comment by tpopela** [comment]
> Actually, this is my work computer, which I use for virtual meetings. No sound is a showstopper so I'm rebasing back to F33. I can't be wrapped up in back-and-forth between upstream and the distribution packagers. 

I suggested to use the upstream tracker as the downstream and upstream maintainer is the same person.

> Can you at least tell me how to opt out of pipewire using fedora 34, or offer something to rebase against which does not use it?

I found this - https://fedoraproject.org/wiki/Common_F34_bugs#Audio_may_not_work_after_upgrade_to_Fedora_34_if_pipewire_was_previously_installed - but I don't know whether it's the same issue.

**Comment by bnordgren** [comment]
So, I think I may well be screwing my system up more. 

I did the advice on #158 and successfully rebased back to F33. Unfortunately, while I now have sound, I have no ability to compile the v4l2loopback kernel module due to the fact that the kernel packages in the toolbox do not match the running silverblue system. #167. I did "pin" this F33 deployment so that it won't disappear. 

I checked out your link. I do not remember installing pipewire previously, but I did have an /etc/pipewire directory (possibly from the F34 upgrade...). I deleted it completely and rebased again to F34, as  Silverblue does not permit the use of `dnf reinstall..` I figured "fake it" with a new rebase. No Joy. Now instead of just a broken server, commands like `pw-cli ls` segfault. I've been slowly repopulating that directory with "conf" files from my working system and rebooting. 

Hah. Copying the *.conf files that were missing from /etc/pipewire and it's child directory seems to have done the trick. I now have sound. 

I have no idea how one would fix this on a Silverblue system if you didn't have a "clean" set of *.conf files to copy over.

**Comment by tpopela** [comment]
> I do not remember installing pipewire previously

Actually PipeWire was preinstalled even before as it was used for screen sharing in WebRTC - it just wasn't used as a PulseAudio replacement.

> I have no idea how one would fix this on a Silverblue system if you didn't have a "clean" set of *.conf files to copy over.

I think that these are always available in /usr/etc/pipewire . You might want to look at https://github.com/ostreedev/ostree/pull/2355/files (I don't know whether it's actually deployed).

**Comment by sebhoss** [comment]
I did a rebase from 33 to 34 a while ago and had no audio problems so far, at least until today. I'm not sure what changed, but here is what I did to get it working again:

```
$ sudo mkdir /etc/pipewire
$ sudo cp -r /usr/share/pipewire/* /etc/pipewire
$ systemctl --user restart pipewire pipewire-pulse
```

There was neither `/etc/pipewire` nor `/usr/etc/pipewire` on my 34 system. According to systemctl all units were active (running) and there was no visible error in journalctl even before copying the config files.

**Comment by roired** [comment]
Good day,

This is still not working, not even the @sebhoss tip, on Silverblue Rawhide [35] Kinoite. No audio and no devices present. In my case, was a fresh install of Silverblue Kinoite Rawhide [35] install.

Regard,
RR

**Comment by sebhoss** [comment]
I actually removed the `/etc/pipewire` folder again after an update to a more recent version of pipewire some time ago. It did work fine until I ran another update this morning. My problem today was that my speakers (which were powered off) were selected as the audio output and not my headphones. I saw that while running `pw-top` and fixed that by simply restarting pipewire with `systemctl --user restart pipewire pipewire-pulse`
